postgres/cloudpostgres/cloudpostgres.go

```diff
@@ -55,6 +61,49 @@ type Params struct {
 	TraceOpts []ocsql.TraceOption
 }
 
+// URLOpener opens GCP PostgreSQL URLs
+// like "cloudpostgres://user:password@myproject.us-central1.instanceId/mydb".
+type URLOpener struct {
+	// TraceOpts contains options for OpenCensus.
+	TraceOpts []ocsql.TraceOption
+}
+
+// Scheme is the URL scheme cloudpostgres registers its URLOpener under on
+// postgres.DefaultMux.
+const Scheme = "cloudpostgres"
+
+func init() {
+	postgres.DefaultURLMux().RegisterPostgres(Scheme, &URLOpener{})
+}
+
+// OpenPostgresURL opens a new GCP database connection wrapped with OpenCensus instrumentation.
+func (uo *URLOpener) OpenPostgresURL(ctx context.Context, u *url.URL) (*sql.DB, error) {
+	vals := u.Query()
+	u.RawQuery = vals.Encode()
+	creds, err := gcp.DefaultCredentials(ctx)
+	client, err := gcp.NewHTTPClient(http.DefaultTransport, creds.TokenSource)
+	certSource := cloudsql.NewCertSource(client)
+	gcpInstance := strings.Split(u.Host, ".")
+	if len(gcpInstance) < 3 {
+		return nil, fmt.Errorf("cloudpostgres: OpenPostgresURL: not enough arguments in GCP host $s", u.Host)
+	}
+	projectId, region, instanceId := gcpInstance[0], gcpInstance[1], gcpInstance[2]
+	password, _ := u.User.Password()
+	params := Params{
```

Comment:
nit: Since this will immediately be used in a pointer context, take the address of the composite literal.

```suggestion
	params := &Params{
```